packages/subcommands/src/lib/utils/SubcommandsHandler.ts

```diff
@@ -0,0 +1,156 @@
+import type { Command } from '../structures/Command';
+import { SubcommandNotLoadedError } from '../errors/SubcommandNotLoadedError';
+import {
+	Args,
+	CommandStore,
+	IPreconditionContainer,
+	PreCommandRunPayload,
+	PreconditionContainerArray,
+	PreconditionContainerSingle
+} from '@sapphire/framework';
+
+export type ResolveSubcommandPayload = Omit<PreCommandRunPayload, 'context'>;
+
+export interface ResolveSubcommandResult {
+	/**
+	 * Subcommand arguments
+	 * @since 1.0.0
+	 */
+	args: Args;
+
+	/**
+	 * Resolved subcommand. Returns undefined if not
+	 * @since 1.0.0
+	 */
+	subcommand: Command | undefined;
+}
+
+export interface SubcommandsPluginOptions {
+	/**
+	 * Continue execution of parent command if command is disabled or error occurred
+	 * @since 1.0.0
+	 * @default false
+	 */
+	continueParent?: boolean;
+
+	/**
+	 * Run parent command when subcommand execution was ended
+	 * @since 1.0.0
+	 * @default false
+	 */
+	runParentAfterSubcommand?: boolean;
+
+	/**
+	 * Preconditions that will be overlap in parents by their subcommands
+	 * @since 1.0.0
+	 * @default []
+	 */
+	overlappedPreconditions?: string[];
+}
+
+export const enum SubcommandsEvents {
+	PreSubcommandRun = 'preSubcommandRun',
+	SubcommandDenied = 'subcommandDenied',
+	SubcommandAccepted = 'subcommandAccepted',
+	SubcommandRun = 'subcommandRun',
+	SubcommandSuccess = 'subcommandSuccess',
+	SubcommandError = 'subcommandError',
+	SubcommandFinish = 'subcommandFinish'
+}
+
+export class SubcommandsHandler {
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	public readonly options: Required<SubcommandsPluginOptions>;
+
+	/**
+	 * @since 1.0.0
+	 * @param options Optional Subcommands plugin settings.
+	 */
+	public constructor(options: SubcommandsPluginOptions = {}) {
+		this.options = {
+			continueParent: options.continueParent ?? false,
+			runParentAfterSubcommand: options.runParentAfterSubcommand ?? false,
+			overlappedPreconditions: options.overlappedPreconditions ?? []
+		};
+	}
+
+	/**
+	 * Resolves subcommand by passed payload.
+	 * @param payload The params that need for resolving subcommand.
+	 */
+	public async resolveSubcommand({ command, message, parameters }: ResolveSubcommandPayload): Promise<ResolveSubcommandResult> {
```

Comment:
Leaving this here from my part in the Discord conversation so it doesn't get lost to time and we can further consider implementations:
"that feels, weird
just register to substores :thinking:
doing it that way would create the need for the store to look up subcommands and ensure they're only used as subcommands too
whereas if you have a second command type (CommandGroup) that acts as a sub store you can just register the commands to and have it act like that, voila
(you could also use the run() block of a CommandGroup to set the default behaviour if there's no subcommand)"

Reply:
By my opinion not need to create one else store or a level of abstraction for commands. And we picked the other way of subcommands realisation in discord if I understood correctly.